web/apps/serve.go

```diff
@@ -57,6 +59,27 @@ func onboarding(c echo.Context) bool {
 	return c.QueryParam("registerToken") != ""
 }
 
+// handleIntent will allow iframes from another app if the current app is
+// opened as an intent
+func handleIntent(c echo.Context, i *instance.Instance, slug, intentID string) {
+	intent := &intents.Intent{}
+	if err := couchdb.GetDoc(i, consts.Intents, intentID, intent); err != nil {
+		return
+	}
+	allowed := false
+	for _, service := range intent.Services {
+		if slug == service.Slug {
+			allowed = true
+		}
+	}
+	if !allowed {
+		return
+	}
+	from := i.SubDomain(intent.Client).Host
+	hdr := fmt.Sprintf("%s %s", middlewares.XFrameAllowFrom, from)
+	c.Response().Header().Set(echo.HeaderXFrameOptions, hdr)
+}
+
```

Comment:
No, you just open the iframe on the new app. `X-Frame-Options` is set on the HTML page inside the iframe and control if the origin of the parent.

For example, you are in the contacts app and you want to pick a photo for an avatar of one contact. You will make a request to the stack to create the intent. In the response, you will have an href for a service. You can then open the iframe on this URL.